src/EnergyPlus/IceThermalStorage.cc

```diff
@@ -1109,15 +1096,20 @@ namespace IceThermalStorage {
                 ErrorsFound = true;
             }
 
-            std::string dischargeCurveType = CurveManager::PerfCurve(DetIceStor(IceNum).DischargeCurveNum).ObjectType;
-            if (dischargeCurveType == "Curve:QuadraticLinear" && cAlphaArgs(5) == "QUADRATICLINEAR") {
-                DetIceStor(IceNum).DischargeCurveTypeNum = CurveQuadraticLinear;
-            } else if (dischargeCurveType == "Curve:CubicLinear" && cAlphaArgs(5) == "CUBICLINEAR") {
-                DetIceStor(IceNum).DischargeCurveTypeNum = CurveCubicLinear;
+            int dischargeCurveDim = CurveManager::PerfCurve(DetIceStor(IceNum).DischargeCurveNum).NumDims;
+            if (dischargeCurveDim == 2 && cAlphaArgs(5) == "FRACTIONCHARGEDLMTD") {
+                DetIceStor(IceNum).DischargeCurveTypeNum = CurveVarsFracChargedLMTD;
+            } else if (dischargeCurveDim == 2 && cAlphaArgs(5) == "FRACTIONDISCHARGEDLMTD") {
+                DetIceStor(IceNum).DischargeCurveTypeNum = CurveVarsFracDischargedLMTD;
+            } else if (dischargeCurveDim == 2 && cAlphaArgs(5) == "LMTDMASSFLOW") {
+                DetIceStor(IceNum).DischargeCurveTypeNum = CurveVarsLMTDMassFlow;
+            } else if (dischargeCurveDim == 2 && cAlphaArgs(5) == "LMTDFRACTIONCHARGED") {
+                DetIceStor(IceNum).DischargeCurveTypeNum = CurveVarsLMTDFracCharged;
             } else {
-                ShowSevereError(cCurrentModuleObject + ": Discharge curve type not valid, type=" + cAlphaArgs(5));
+                ShowSevereError(cCurrentModuleObject + ": Discharge curve independent variable options not valid, option=" + cAlphaArgs(5));
```

Comment:
If I'm following correctly, the user could select a valid choice for variable type, but specify a curve with the wrong dimensions and this error message would fire. It would be cleaner to separate these two and have separate error messages for invalid variable options or invalid curve dimension. Same for the charge curve.

Reply:
@mjwitte Now separated, for both discharging and charging.